openwisp_monitoring/check/classes/base.py

```diff
@@ -0,0 +1,45 @@
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from swapper import load_model
+
+from openwisp_controller.config.models import Device
+
+Check = load_model('check', 'Check')
+Metric = load_model('monitoring', 'Metric')
+
+
+class BaseCheck(object):
+    def validate_instance(self):
+        # check instance is of type device
+        obj = self.related_object
+        if not obj or not isinstance(obj, Device):
+            message = 'A related device is required to perform this operation'
+            raise ValidationError({'content_type': message, 'object_id': message})
+
+    def validate(self):
+        self.validate_instance()
+        self.validate_params()
+
+    def validate_params(self):
+        pass
+
```

Comment:
I forgot something else, which is just something which is usually done when developing these kind of object oriented patterns:

```python
def check(self, store=True):
    raise NotImplementedError
```

It's usage is recommended in the python docs, I suggest to read this passage: https://docs.python.org/3/library/exceptions.html#NotImplementedError